apps/dashboard/src/components/contract-components/contract-deploy-form/modular-contract-default-extensions-fieldset.tsx

```diff
@@ -0,0 +1,106 @@
+import { FormControl } from "@chakra-ui/react";
+import { useQuery } from "@tanstack/react-query";
+import { SolidityInput } from "contract-ui/components/solidity-inputs";
+import { getExtensionInstalledParams } from "contract-ui/tabs/manage/components/getExtensionInstalledParams";
+import invariant from "tiny-invariant";
+import { FormErrorMessage, FormLabel, Heading } from "tw-components";
+import type { CustomContractDeploymentForm } from "./custom-contract";
+
+type ExtensionMeta = {
+  extensionName: string;
+  extensionVersion: string;
+  publisherAddress: string;
+};
+
+/**
+ * Get the install params for the "encodeBytesOnInstall" function for all given extensions for a given modular contract
+ * to render the form in deploy contract form
+ */
+export function useModularContractsDefaultExtensionsInstallParams(props: {
+  defaultExtensions?: ExtensionMeta[];
+  isQueryEnabled: boolean;
+}) {
+  const { defaultExtensions, isQueryEnabled } = props;
+  return useQuery(
+    ["useModularContractsDefaultExtensionsInstallParams", defaultExtensions],
+    async () => {
+      invariant(defaultExtensions, "defaultExtensions must be defined");
+
+      const promises = defaultExtensions.map(async (ext) => {
+        return getExtensionInstalledParams(ext);
+      });
+
+      return Promise.all(promises);
+    },
+    {
+      enabled: !!(isQueryEnabled && defaultExtensions),
+      refetchOnWindowFocus: false,
+    },
+  );
+}
```

Comment:
please use the modern way to write queries
```ts
useQuery({
  queryKey: [...],
  queryFn: async () => {...},
  ...
});
```

Reply:
hmm, weird - I thought the version we were using in dashboard did not support this syntax. looks like it works - updating..